packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -249,10 +257,27 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             // ODSP document ids are random guids (different per session)
             // fix the branch name in attributes
             // this prevents issues when generating summaries
-            const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
-            documentAttributes.branch = this.documentId;
+            if (blob.encoding === "base64") {
+                const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
```

Comment:
I do not remember what state of overall work it is, but we should stop assuming that all encoding is base64, we should respect blob.encoding. And I'd think it should be first change, mostly because it unblocks server to start sending data without base64 encoding.